lib/install/validate/compute.go

```diff
@@ -42,6 +42,47 @@ func (v *Validator) compute(op trace.Operation, input *data.Data, conf *config.V
 	// TODO: for RP creation assert whatever we decide about the pool - most likely that it's empty
 }
 
+func (v *Validator) checkVMGroup(op trace.Operation, input *data.Data, conf *config.VirtualContainerHostConfigSpec) {
+	defer trace.End(trace.Begin("", op))
+
+	if input.UseVMGroup {
+		if !v.IsVC() {
+			v.NoteIssue(errors.New("DRS VM Groups may only be configured when using VC"))
+			return
+		}
+
+		if v.Session.DRSEnabled == nil || !*v.Session.DRSEnabled {
+			v.NoteIssue(errors.New("DRS VM Groups may not be used without DRS"))
+			return
+		}
+
+		if input.ID != "" {
+			op.Debug("Skipping DRS VM Group existence check as VCH has already been created")
+			return
+		}
+
+		conf.UseVMGroup = input.UseVMGroup
+		// For now, we always name the VM Group based on the name of the VCH
+		conf.VMGroupName = conf.Name
+
+		if v.Session.Cluster == nil {
+			// We already note a more helpful issue for this following the compute method's call to ResourcePoolHelper.
+			v.NoteIssue(errors.New("Unable to determine presence of DRS VM Groups due to previous errors"))
+			return
+		}
+
+		exists, err := VMGroupExists(op, v.Session.Cluster, conf.VMGroupName)
```

Comment:
As a note, most of the other items such as this rely on vSphere to reject them if there's already something of that name.
For VMgroups I think vSphere returns InvalidArgument or DuplicateName, I don't recall which.

Reply:
To recap our in-person discussion for anyone reading this PR: we should eventually separate validation of the requested _configuration_ from validation of the _environment_. Unlike most existing validation logic, this check belongs with the latter category. For now, it's desirable to keep this check where it is to allow the creation process to fail fast (without it, we would finish creating the endpoint VM before attempting the create the VM Group and discovering the naming collision). 

---

It looks like we also handle a rejection at creation time, which should cover the case where a VM Group is created after this existence check.

https://github.com/vmware/vic/blob/22b228aea32b8174449c9117ff6b43d2843154da/lib/install/management/vmgroup.go#L53-L55

(Note that this uses `IsTransientError`, which does not retry on `InvalidArgument`.)

https://github.com/vmware/vic/blob/22b228aea32b8174449c9117ff6b43d2843154da/lib/install/management/create.go#L102-L104